src/main/scala/com/databricks/labs/overwatch/pipeline/Restore.scala

```diff
@@ -0,0 +1,126 @@
+  private val workSpace = _workspace
+  private val sourceETLDB = _sourceETLDB
+  private val bronze = Bronze(workSpace)
+  private val silver = Silver(workSpace)
+  private val gold = Gold(workSpace)
+
+  private val logger: Logger = Logger.getLogger(this.getClass)
+  val Config = _config
+
+  private[overwatch] def restore(
+                                  allTarget: Array[PipelineTable],
+                                  cloneLevel: String = "DEEP",
+                                ): Unit = {
+    val acceptableCloneLevels = Array("DEEP", "SHALLOW")
+    require(acceptableCloneLevels.contains(cloneLevel.toUpperCase), s"SNAP CLONE ERROR: cloneLevel provided is " +
+      s"$cloneLevel. CloneLevels supported are ${acceptableCloneLevels.mkString(",")}.")
+
+    try {
+      dbutils.fs.ls(targetPrefix)
+    } catch {
+      case e: Exception => dbutils.fs.mkdirs(targetPrefix)
+    }
+
+    try {
+      if (!dbutils.fs.ls(targetPrefix).nonEmpty) {
+        val cloneSpecs = new Snapshot(workSpace.getConfig.databaseName, targetPrefix, workspace, workspace.database, workspace.getConfig, "restore").buildCloneSpecs(cloneLevel, allTarget)
+        val cloneReport = Helpers.parClone(cloneSpecs)
+        val restoreReportPath = s"${targetPrefix}/restore_report/"
+        cloneReport.toDS.write.format("delta").mode("append").save(restoreReportPath)
+      } else {
+        println("Target Path is not Empty...... Could not proceed with Restoration")
+      }
+
+    } catch {
+      case e: Throwable =>
+        val failMsg = PipelineFunctions.appendStackStrace(e)
+        logger.log(Level.WARN, failMsg)
+    }
+  }
+}
+
+object Restore extends SparkSessionWrapper {
+  private val logger: Logger = Logger.getLogger(this.getClass)
+
+    def apply(
+               workSpace: Workspace,
+               allTarget: Array[PipelineTable],
+               targetPrefix : String,
+               backupPath: String,
+               CloneLevel: String
+             ): Any = {
+
+      val restoration = new Restore(workSpace.getConfig.databaseName, targetPrefix, workSpace, workSpace.database, workSpace.getConfig)
+      restoration.restore(allTarget,CloneLevel)
+
+    }
+
+    /**
+     * Create a backup of the Overwatch datasets
+     *
+     * @param arg(0)        Source ETL Path Prefix from where restore need to be performed
+     * @param arg(1)        Target ETL Path Prefix to where restore data would be loaded.
+     * @return
+     */
+
+    def main(args: Array[String]): Unit = {
+
+      val sourcePrefix = args(0)
+      val orgID = Initializer.getOrgId
+      val targetPrefix = args(1)
+      val cloneLevel = "Deep"
+      //
+
+
+      val backupPath = s"${sourcePrefix}/data"
+      val pipReportPath = s"${backupPath}/pipeline_report"
+      try{
+        dbutils.fs.ls(s"$pipReportPath/_delta_log").nonEmpty
+        logger.log(Level.INFO, s"Overwatch has being deployed with ${pipReportPath} location...proceed")
```

Comment:
we're not deploying, I think this text would confuse users

Reply:
Done